libraries/botframework-streaming/tests/NamedPipe.test.js

```diff
@@ -414,5 +415,29 @@ describe('Streaming Extensions NamedPipe Library Tests', () => {
             expect(server.disconnect()).to.not.throw;
             done();
         });
+
+        
+        it('calling createNodeServer() should throw if passing in a callback that\'s not a function', () => {
+            const stringCallback = 'Not a real callback function.';
+            expect(() => NodeServerUtils.createNodeServer(stringCallback)).to.throw;
+        });
+
+        it('should not throw when choosing not to pass in a callback at all into createNodeServer()', () => {
+            expect(() => NodeServerUtils.createNodeServer()).to.not.throw;
+        });
+
+        it('should return a Server when calling createNodeServer()', () => {
+            const server = NodeServerUtils.createNodeServer();
+
+            expect(server).to.not.be.null;
+            expect(server).to.be.instanceOf(Object);
+        });
+
+        it('should return the constructor when calling getNodeServerConstructor()', () => {
+            const netServerCtor = NodeServerUtils.getNetServerConstructor();
+
+            expect(netServerCtor).to.not.be.null;
+            expect(typeof netServerCtor).to.equal('function');

```

Comment:
maybe use a `is` assertion function from TypeScript to assert that the function returns the right type?

``` typescript
function IsNetServer(o: any): o is Server {
    if(o.on && typeof o.on === 'function') {
        return true;
    } // or whatever the right logic is?
}
```

Reply:
Added server type guards into `createNodeServer()`, as the tests are in plain JS.
Test that they don't throw with the typeguards in place